### PR TITLE
Restructuring and reconnection improvement

### PR DESCRIPTION
**Pull request by rhoegg**
This restructures the connector to make the operations reuse common connectivity code.

- Enable [reconnection](https://docs.mulesoft.com/mule-sdk/1.1/reconnection) by using ConnectionException for OAuth token expiration. This was found when incorporating the connector in long running processes.
- Move errors to internal package, and other updates found in [Error Handling best practices](https://docs.mulesoft.com/mule-sdk/1.1/error-handling)
- Extract common code for Azure Key Vault API requests and responses to AzureKeyVaultClient and KeyVaultRequest classes.

### CODE REVIEW

**Comment by rhoegg** [comment]
Resolves #23

File: src/main/java/com/avioconsulting/mule/connector/akv/provider/internal/error/AzureKeyVaultErrorType.java [hunk 1]
```diff
@@ -7,5 +7,5 @@
   SECRET_NOT_FOUND,
   KEY_NOT_FOUND,
   CERTIFICATE_NOT_FOUND,
-  UNKNOWN_ERROR
```
Comment (kkingavio) [inline src/main/java/com/avioconsulting/mule/connector/akv/provider/internal/error/AzureKeyVaultErrorType.java:10]: Why are we removing the UNKNOWN_ERROR?

File: src/main/java/com/avioconsulting/mule/connector/akv/provider/api/model/OAuthError.java [hunk 2]
```diff
@@ -16,7 +16,7 @@
   private List<Integer> errorCodes;
 
   @SerializedName(value = "timestamp")
-  private Date timestamp;
+  private String timestamp;
```
Comment (kkingavio) [inline src/main/java/com/avioconsulting/mule/connector/akv/provider/api/model/OAuthError.java:19]: Is there a good reason to change this from a Date to a String?

File: src/main/java/com/avioconsulting/mule/connector/akv/provider/internal/error/AzureKeyVaultErrorType.java (hunk 1 above)
Comment (rhoegg) [reply src/main/java/com/avioconsulting/mule/connector/akv/provider/internal/error/AzureKeyVaultErrorType.java:10]: This was due to Mulesoft's best practices here: https://docs.mulesoft.com/mule-sdk/1.1/error-handling

File: src/main/java/com/avioconsulting/mule/connector/akv/provider/api/model/OAuthError.java (hunk 2 above)
Comment (rhoegg) [reply src/main/java/com/avioconsulting/mule/connector/akv/provider/api/model/OAuthError.java:19]: I got a deserialization when manually testing with the Microsoft API. This fixed it.

**Review by kkingavio — APPROVED** [review]
Thanks for contributing!

**Pull request merged**